src/pages/Admin/Charity/Permissions/MobileTable.tsx

```diff
@@ -0,0 +1,127 @@
+import { Disclosure } from "@headlessui/react";
+import { useFormContext } from "react-hook-form";
+import { useAdminResources } from "pages/Admin/Guard";
+import { useGetWallet } from "contexts/WalletContext";
+import { DrawerIcon } from "components/Icon";
+import { CheckField, Field } from "components/form";
+import { FormValues } from "./schema";
+
+export default function MobileTable({ className = "", disabled = false }) {
+  const { endow_type } = useAdminResources<"charity">();
+  const { wallet } = useGetWallet();
+  const { getValues, register, watch } = useFormContext<FormValues>();
+
+  const { initialValues, ...formValues } = getValues();
+
+  return (
+    <div className={`${className} border border-prim rounded-t`}>
+      <div className="grid items-center grid-cols-[auto_1fr] h-12 uppercase text-xs font-bold bg-orange-l6 dark:bg-blue-d7 border-b border-prim divide-x divide-prim rounded-t">
+        <div className="w-12" />
+        <div className="py-3 px-4">Action</div>
+      </div>
+
+      {(
+        Object.keys(formValues) as (keyof Omit<FormValues, "initialValues">)[]
+      ).map((formField) => (
+        <Disclosure
+          key={`mob-table-row-${formField}`}
+          as="div"
+          className="text-sm odd:bg-orange-l6 dark:even:bg-blue-d6 dark:odd:bg-blue-d7 w-full border-b last:border-0 border-prim"
+        >
+          {({ open }) => {
+            const modifiable = watch(`${formField}.modifiable`);
+            const delegate = watch(`${formField}.delegate`);
+            const initDelegate = watch(`initialValues.${formField}.delegate`);
+            const init_delegate_address = watch(
+              `initialValues.${formField}.delegate_address`
+            );
+            const isDisabled =
+              !modifiable ||
+              disabled ||
+              (initDelegate &&
+                !!init_delegate_address &&
+                init_delegate_address !== wallet?.address);
```

Comment:
whether connected wallet has control been deligated to shouldn't affect how he would interact to individual settings, nor `modifiable` in the `whole` endowment controller context

NOTE: this is with the execption for `EndowmentController.endowment_controller` setting, for which if delegated to a single entity  `Z` would mean `Mr.Z` is the only one who could modify all other individual settings

consider this scenario
1. initialy, `owner` ( whether single wallet or multisig wallet say `AdminX` ) is allowed to modify `EndowmentController`  as specified in `EndowmentController.endowment_controller` field which is defaulted to 
```rust
pub fn default(endow_type: &EndowmentType) -> Self {
        SettingsPermissions {
            owner_controlled: true,
            gov_controlled: false,
            modifiable: match endow_type {
                EndowmentType::Charity => false,
                _ => true,
            },
            delegate: None,
        }
    }
```

2. let's say `A`, `B`, `C` are members of `AdminX`. and `A` creates a proposal to delegate all inividual settings to `A` and set all fields to `not modifiable`

```ts
//PERMISSIONS 
1.`fees` --> delegate --> `A`
2.`beneficiaries` --> delegate --> `A`
3. `contributors` --> delegate --> `A`
```
and everyone approves, making the changes effective

3.  this would mean that `B, C` could not anymore modify permissions `1`, `2`, `3`, though technically they should be able to since they are still a member of the `owner` specified in `Endomwent.endowment_controller`

Reply:
@ap-justin this is related to the comment above https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#discussion_r1136474481 ?

Reply:
Waiting on Andrey to confirm the exact requirements here, see https://github.com/AngelProtocolFinance/angelprotocol-web-app/pull/1885#issuecomment-1469530382.

Reply:
> whether connected wallet has control been deligated to shouldn't affect how he would interact to individual settings, nor `modifiable` in the `whole` endowment controller context
> 
> NOTE: this is with the execption for `EndowmentController.endowment_controller` setting, for which if delegated to a single entity `Z` would mean `Mr.Z` is the only one who could modify all other individual settings
> 
> consider this scenario
> 
> 1. initialy, `owner` ( whether single wallet or multisig wallet say `AdminX` ) is allowed to modify `EndowmentController`  as specified in `EndowmentController.endowment_controller` field which is defaulted to
> 
> ```rust
> pub fn default(endow_type: &EndowmentType) -> Self {
>         SettingsPermissions {
>             owner_controlled: true,
>             gov_controlled: false,
>             modifiable: match endow_type {
>                 EndowmentType::Charity => false,
>                 _ => true,
>             },
>             delegate: None,
>         }
>     }
> ```
> 
> 2. let's say `A`, `B`, `C` are members of `AdminX`. and `A` creates a proposal to delegate all inividual settings to `A` and set all fields to `not modifiable`
> 
> ```ts
> //PERMISSIONS 
> 1.`fees` --> delegate --> `A`
> 2.`beneficiaries` --> delegate --> `A`
> 3. `contributors` --> delegate --> `A`
> ```
> 
> and everyone approves, making the changes effective
> 
> 3. this would mean that `B, C` could not anymore modify permissions `1`, `2`, `3`, though technically they should be able to since they are still a member of the `owner` specified in `Endomwent.endowment_controller`

@SovereignAndrey can you comment on this flow, seems like an important (potential) edge case?

Reply:
> @SovereignAndrey can you comment on this flow, seems like an important (potential) edge case?

@misicnenad @ap-justin This is not an issue, as in the above scenario **no one** would be able to modify the `EndowmentSettings` values for the `fees, beneficiaries, or contributors` once the proposal passed, since that proposal has made those fields **not** modifiable. The modifiable check if done first before ownership & message sender ID checks, so it applies regardless of whomever might have/had the ability to change a given field/attribute. Making `A` a delegate in the same proposal was a non-event since they were immediately locked out at the same time.